### DISCUSSION TITLE
Is there setup documentation?

### DISCUSSION THREAD

**Discussion by bjageman** [AdmiralGT/botc-scripts#423] [Q&A] [answered]
I'm trying to get this app to run on my local server and there's not a lot to go on for setting this up. I know a lot of this is typical Django knowledge but it would be nice to have some info on:

- Python dependencies
- What's the bare minimum to set up your dev and production environment
- Initially population your data on the site

I eventually figured it out but I'm currently stuck on the last part and can't seem to figure this out. My version doesn't have any of the characters listed and so I'm unable to upload files because an error message shows "Script must contain at least 14 official Blood on the Clocktower character"

**Comment by AdmiralGT** [comment, marked as answer]
Sorry about the complete lack of documentation, I appreciate that it's really frustrating there not being any and I need to do better here. It's been on my backlog for a while and I've just never invested the time to create any as I didn't expect to get many people wanting to contribute or play around with the code. I'll put some time in to improving that so if you have any input for what would be good to include in that it would be great to hear where your pain points were.

To get around the validation issue there are 2 approaches. The simplest would be to comment out the validator, line 46-48 in scripts/validators.py. That will still leave a bunch of wonkyness around the site but will let you add scripts and have some data.

The other method would be to create a superuser `python manage py createsuperuser`, login to the admin site `localhost:8080/admin` and then create characters through the GUI.

I'll see if I can pull together some scripts that can be run to populate the characters and add some basic scripts to help out getting started with local development as well as the docs.

**Comment by bjageman** [comment]
Thanks for the speedy reply! I'd actually love to contribute if you'd like the help. I'm out of work and could use a project to keep my skills fresh. I also have a large BOTC group and we've been trying to find a good way to share our custom scripts with each other so this app is a huge help. 

I can put up a PR for a suggested README doc if that works. I also set up the app to work on Docker so I can provide another PR with the Dockerfile and docker-compose.yml.
